crates/broker/src/lib.rs

```diff
@@ -203,8 +213,25 @@ struct Order {
     error_msg: Option<String>,
 }
 
+// An Order represents a proof request and a specific method of fulfillment.
+//
+// Requests can be fulfilled in multiple ways, for example by locking then fulfilling them,
+// by waiting for an existing lock to expire then fulfilling for slashed stake, or by fulfilling
+// without locking at all.
+//
+// For a given request, each type of fulfillment results in a separate Order being created, with different
+// FulfillmentType values.
+//
+// Additionally, there may be multiple requests with the same request_id, but different ProofRequest
+// details. Those also result in separate Order objects being created.
+//
+// See the id() method for more details on how Orders are identified.
```

Comment:
nit: This likely should be on the `Order` struct and be rustdoc comments (`///`)